### PR TITLE
TST: update Travis config to use container infrastructure.

### PR DESCRIPTION
**Pull request by arokem**
See Travis docs for details:
http://docs.travis-ci.com/user/migrating-from-legacy/?utm_source=legacy-notice&utm_medium=banner&utm_campaign=legacy-upgrade

### CODE REVIEW

**Comment by arokem** [comment]
Just thought I would remind the others about this. It's a two-line change that will make our travis runs go faster. Pure win. Please review this.

File: .travis.yml [hunk 1]
```diff
@@ -32,7 +33,6 @@ before_install:
     - source venv/bin/activate
     - python --version # just to check
     - retry pip install nose # always
-    - sudo apt-get install libblas-dev liblapack-dev libatlas3gf-base
```
Comment (matthew-brett) [inline .travis.yml]: I think you need to add something like:

```
cache: pip
addons:
  apt:
    packages:
    - ibblas-dev
    - liblapack-dev
    - libatlas3gf-base
```

https://github.com/scikit-image/scikit-image/pull/1607

Comment (arokem) [inline .travis.yml]: Why? As far as I can tell, everything works just fine without these packages.

Comment (matthew-brett) [inline .travis.yml]: OK - maybe these get installed with the numpy installations.

But - I think you will need some sort of pip cacheing: https://github.com/scikit-image/scikit-image/pull/1610

Comment (arokem) [inline .travis.yml]: Like so?

**Comment by matthew-brett** [comment]
OK - let's give it a shot...

**Pull request merged**